File: Benchmarking/noscaling/euler47/euler47.java
import java.util.*;
import java.math.*;

public class euler47
{
	public static void main(String[] args)
	{
		int cons = Integer.parseInt(args[0]);
		int dist = Integer.parseInt(args[1]);
		calc(cons, dist);
	}
	
	private static List<Integer> prime_factors(int n)
	{
		List<Integer> factors = new ArrayList<>();
		int d = 2;
		while(n > 1)
		{
			while(n % d == 0)
			{
				factors.add(d);
				n = n / d;
			}
			d += 1;
		}
		return factors;
	}
	
	private static int calc(int cons, int dist)
	{
		long m = 9223372036854775807l;
		int currentCons = 0;
		List<Integer> arr;
		for(int i = 1; i <= m; i++)
		{
			arr = new ArrayList<>(new LinkedHashSet<>(prime_factors(i)));
			if(arr.size() == dist)
			{
				currentCons += 1;
				if(currentCons == cons)
				{
					return i - (cons - 1);
				}
			}
			else
			{
				currentCons = 0;
			}
		}
		return 0;
	}
}

File: Benchmarking/noscaling/euler47/euler47.jl
#=
The first two consecutive numbers to have two distinct prime factors are:

14 = 2 × 7
15 = 3 × 5

The first three consecutive numbers to have three distinct prime factors are:

644 = 2² × 7 × 23
645 = 3 × 5 × 43
646 = 2 × 17 × 19.

Find the first four consecutive integers to have four distinct prime factors. What is the first of these numbers?
=#
#Answer: 134043

#Terminal: julia euler47.jl 4 4
cons = parse(Int, ARGS[1])
dist = parse(Int, ARGS[2])

function prime_factors(n)
  factors = Int[]
  d = 2
  while n > 1
    while n % d == 0
      push!(factors, d)
      n = div(n, d)
    end
    d += 1
  end
  return factors #array of prime factors of n
end

function calc(cons, dist)
  m = typemax(Int64)
  currentCons = 0
  for i = 1 : m
    arr = unique(prime_factors(i)) #Removes dublicates in the returned array from prime_factors
    if length(arr) == dist
      currentCons += 1
      if currentCons == cons
        return i - (cons - 1)
      end
    else
      currentCons = 0
    end
  end
  return "$cons consecutive numbers was not found with $dist distinctive factors."
end

calc(cons, dist)
